libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/TxGen.hs

```diff
@@ -677,27 +677,33 @@ genRecipientsFrom txOuts = do
           v = getTxOutVal reify txout
       genWithChange s txout rs = do
         let !(!addr, !v, !ds) = txoutFields reify txout
-        c <- Coin <$> lift (choose (1, unCoin(coin v)))  -- v could be 0
```

Comment:
I guess this was the bug? if `v` is zero, are we accidentally "partitioning" zero into `{ 1 }`?

Reply:
Yes, my original assumption was that there could never be a TxOut with `0` ada in it, which is the case for Cardano mainnet, but turns out in theory with specific PParams, it is a legitimate state, which indeed was triggering this bug in the generator.

So when such TxOut is present in the inputs, this would result in creation a TxOut in a transaction of one more lovelace that there was in the inputs. That is because `choose (1, 0) === choose (0, 1)`.

Reply:
It's such a great feeling for everything to make sense again, great catch! :)